lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1057,12 +1058,21 @@ public void unpublishObjectsExternally(Set<GUID> guids, int accessGroupId) throw
     public List<ObjectData> getObjectsByIds(Set<GUID> ids, PostProcessing pp) 
             throws IOException {
 
-        Map<String, Object> doc = ImmutableMap.of("query",
-                                     ImmutableMap.of("bool",
+        Map<String, Object> query = ImmutableMap.of("bool",
                                         ImmutableMap.of("filter",
                                            ImmutableMap.of("terms",
-                ImmutableMap.of("guid", ids.stream().map(u -> u.toString()).collect(Collectors.toList()))))));
+                ImmutableMap.of("guid", ids.stream().map(u -> u.toString()).collect(Collectors.toList())))));
         //doc.put("_source", Arrays.asList("ojson"));
+        Map<String, Object> highlight =
```

Comment:
This appears to be duplicated in a couple places - DRY it up with a method

Reply:
okay. actually since we are talking about duplicates. Why does fn searchIds even exits. Its the same as searchObjects just with null passed for PostProcessing. The naming of the methods are also super weird. getObjectsByIds is just search but searchIds doesnt even search on ids.

Reply:
No idea, before my time.

Reply:
Looks like they're only used in tests

Reply:
So I looked at all the queries and I could pull out the 2-3 lines that are similar but I feel like it reads less well. The only things staying the same is bool, then filter. What do you think?

Reply:
Hmm, our opinions differ on this one I guess. I think it's easier to read as the highlight query creation is abstracted away. More importantly, if the highlight query needs to change we can do it in one place vs 2+.